features/netsocket/InternetSocket.h

```diff
@@ -117,6 +117,12 @@ class InternetSocket : public Socket {
      */
     virtual void sigio(mbed::Callback<void()> func);
 
+    /** Get the remote-end peer associated with this socket.
+     *
+     *  @return         pointer to the remote_peer SocketAddress
+     */
+    const SocketAddress* getpeername();
```

Comment:
For lifetime reasons and general POSIX compatibility - aligning with the way we've mapped other functions - this should be prototyped as `virtual nsapi_error_t getpeername(SocketAddress *address)`, assuming implementing a pure virtual in `Socket` as suggested.

(Other calls already omit the POSIX length parameter for addresses, on the assumption `SocketAddress` has space for all possible address types...)